emission/tests/modellingTests/TestForestModel.py

```diff
@@ -88,86 +70,64 @@ def tearDown(self):
-    #     else:
-    #         # Compare the current predictions with the previous predictions
-    #         self.assertPredictionsMatch(previous_predictions, current_predictions)
-
-    # def load_previous_predictions(self):
-    #     # Retrieve stored predictions from the database
-    #     # Using get_analysis_timeseries_db as an example, replace with the correct method if needed
-    #     db = edb.get_analysis_timeseries_db()
-    #     predictions = db.find_one({"user_id": self.user_id, "metadata.key": "predictions"})
-    #     return predictions
-
-    # def run_current_model(self):
-    #     # Placeholder: Run the current model and get predictions
-    #     # Replace this with the actual model running code
-    #     predictions = None
-    #     return predictions
-
-    # def store_predictions(self, predictions):
-    #     # Store the predictions in the database
-    #     # Using get_analysis_timeseries_db as an example, replace with the correct method if needed
-    #     db = edb.get_analysis_timeseries_db()
-    #     entry = {
-    #         "user_id": self.user_id,
-    #         "metadata": {
-    #             "key": "predictions",
-    #             "write_ts": pd.Timestamp.now().timestamp()  # Using pandas timestamp as an example
-    #         },
-    #         "data": predictions
-    #     }
-    #     db.insert_one(entry)
-
-    # def assertPredictionsMatch(self, prev, curr):
-    #     # Placeholder: Check if the predictions match
-    #     # This will depend on the format and type of your predictions
-    #     # For example, if predictions are lists or arrays, you can use numpy
-    #     if not np.array_equal(prev, curr):
-    #         self.fail("Current model predictions do not match previously stored predictions!")
+        curr_predictions_list = eamur.predict_labels_with_n(
+            trip_list = [test_data],
+            model=model           
+        )
+
+
+        ## predictions take the form like :
+        #
+        #{'labels': {'mode_confirm': 'ebike', 'replaced_mode': 'walk', 'purpose_confirm': 'dog-park'}, 'p': 1.0}
+        #  we can store these predictions in a json and then for every run other than the first we
+        # can load the predictions and compare
+
+        try:
+            if os.path.exists(file_path) and os.path.getsize(file_path)>0:
+                with open(file_path, 'r') as f:
+                    prev_predictions_list = json.load(f)
+                    logging.debug()
+                    self.assertEqual(prev_predictions_list,curr_predictions_list," previous predictions should match current predictions")
+            else:
+                with open(file_path,'w') as file:
+                    json.dump(curr_predictions_list,file,indent=4)
+                    logging.debug("Previous predicitons stored for future matching" )
+        except json.JSONDecodeError:
+            logging.debug("jsonDecodeErrorError")
```

Comment:
This part is questionable. I wanted to know if dumping ( writing predictions of first run ever to be used as ground truth) in a JSON here is a good practice or not?

 Other option is to store the past predictions  in db. But I think the tear-down will clean it up while exiting. So I opted for JSON. Is there a way to protect the writing. Let me know if I am missing something here.